geonotebook/vis/ktile/ktile.py

```diff
@@ -167,54 +167,81 @@ def ingest(self, data, name=None, **kwargs):
+            # Note:
+            # Check if the reader has defined a vrt_path
+            #
+            # This is mostly intended for the VRTReader so that it can communicate
+            # that the VRT for reading data is also the VRT that should be used for
+            # visualisation. Otherwise we wouild have to explicitly add a vrt_path
+            # kwarg to the add_layer() call.
+            #
+            # A /different/ VRT can still be used for visualisation by passing
+            # a path via vrt_path to add_layer.
+            #
+            # Finally, A dynamic VRT will ALWAYS be generated if vrt_path is
+            # explicitly set to None via add_layer.
+            if hasattr(data.reader, 'vrt_path'):
+                if 'vrt_path' in kwargs and kwargs['vrt_path'] is None:
+                    # Explicitly set to None
+                    pass
+                else:
+                    kwargs['vrt_path'] = data.reader.vrt_path
+
+            # If we have a static VRT
+            if 'vrt_path' in kwargs and kwargs['vrt_path'] is not None:
+                options.update(self._static_vrt_options(data, kwargs))
             else:
-                kwargs['vrt_path'] = data.reader.vrt_path
-
-        # If we have a static VRT
-        if 'vrt_path' in kwargs and kwargs['vrt_path'] is not None:
-            options.update(self._static_vrt_options(data, kwargs))
+                # We don't have a static VRT, set options for a dynamic VRT
+                options.update(self._dynamic_vrt_options(data, kwargs))
+
+            r = requests.post(base_url, json={
+                "provider": {
+                    "class": "geonotebook.vis.ktile.provider:MapnikPythonProvider",
+                    "kwargs": options
+                }
+                # NB: Other KTile layer options could go here
+                #     See: http://tilestache.org/doc/#layers
+            })
         else:
-            # We don't have a static VRT, set options for a dynamic VRT
-            options.update(self._dynamic_vrt_options(data, kwargs))
-
-        # Make the Request
-        base_url = '{}/{}/{}'.format(self.base_url, kernel_id, name)
-
-        r = requests.post(base_url, json={
-            "provider": {
-                "class": "geonotebook.vis.ktile.provider:MapnikPythonProvider",
-                "kwargs": options
-            }
-            # NB: Other KTile layer options could go here
-            #     See: http://tilestache.org/doc/#layers
-        })
+          raise ValueError("Don't know how to ingest specified data")
 
         if r.status_code == 200:
             return base_url
         else:
             raise RuntimeError(
-                "KTile.ingest() returned {} error:\n\n{}".format(
-                    r.status_code, ''.join(r.json()['error'])))
+                "KTile.ingest() returned error {}:\n\n{}".format(
+                    r.status_code, r))
+
+    def disgorge(self, name, **kwargs):
```

Comment:
Disgorge makes sense to me as a method that is missing from the vis_server API. It seems like if vis_server's have a shot at the data on add_layer()  they should get a shot at the data on remove_layer().  With that said,  it would be great if the API's could be similar (e.g.  disgorge taking a ```RasterData``` data object rather than just a name). 

If it took the RasterData object, and the RasterData object had the tms object saved as apart of it's state,  then it should be able to call unbind without having to save state in ```_server_state``` right?